### DISCUSSION TITLE
Enabling-disabling fixed constraints between objects during run-time.

### DISCUSSION THREAD

**Discussion by kvasios** [google-deepmind/mujoco#1208] [General]
Hi, I was wandering whether MuJoCo allows for setting fixed constraints between objects dynamically during run-time as e.g pybullet's add/remove constraint functions. Or any alternatives for mimicking this behavior? Any working example would be appreciated as I haven't found anything along those lines.

**Comment by yuvaltassa** [comment]
Yes. This was already requested in #876 and addressed in ee78b8f76ba20b3d414a8ea1cda2cbe6e9cc96b6 (item 13 in the [3.0.0 release notes](https://mujoco.readthedocs.io/en/stable/changelog.html#version-3-0-0-october-18-2023))

You can see usage examples in the tests, e.g. [here](https://github.com/google-deepmind/mujoco/blob/d3eee0316c87dca14aa7ad890d3dc9ae302fe062/test/engine/engine_forward_test.cc#L558)

Next time, please try to check if what you want was already requested.

**Reply by kvasios** [reply]
Thanks for the response. Quite helpful to have those resources. I've tried a quick search on the issue before posting but I haven't gotten any relevant results.

**Comment by yuvaltassa** [comment]
I'm now realizing the `eq_data` variable should be documented [here](https://mujoco.readthedocs.io/en/stable/computation/index.html#user-inputs) but is currently not mentioned. Reopening to remind us to do this.

**Comment by yuvaltassa** [comment]
Docs on Latest branch [now up to date](https://mujoco.readthedocs.io/en/latest/computation/index.html#user-inputs).
